presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -70,39 +72,43 @@ public void testAcos()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("acos(" + doubleValue + ")", DOUBLE, Math.acos(doubleValue));
-            assertFunction("acos(FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.acos((float) doubleValue));
+            assertFunction("acos(FLOAT '" + (float) doubleValue + "')", FLOAT, (float) Math.acos((float) doubleValue));
         }
-        assertFunction("acos(NULL)", DOUBLE, null);
+        assertFunction("acos(NULL)", FLOAT, null);
+        assertFunction("acos(CAST(NULL AS DOUBLE))", DOUBLE, null);
     }
 
     @Test
     public void testAsin()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("asin(" + doubleValue + ")", DOUBLE, Math.asin(doubleValue));
-            assertFunction("asin(FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.asin((float) doubleValue));
+            assertFunction("asin(FLOAT '" + (float) doubleValue + "')", FLOAT, (float) Math.asin((float) doubleValue));
         }
-        assertFunction("asin(NULL)", DOUBLE, null);
+        assertFunction("asin(NULL)", FLOAT, null);
+        assertFunction("asin(CAST(NULL AS DOUBLE))", DOUBLE, null);
     }
 
     @Test
     public void testAtan()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("atan(" + doubleValue + ")", DOUBLE, Math.atan(doubleValue));
-            assertFunction("atan(FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.atan((float) doubleValue));
+            assertFunction("atan(FLOAT '" + (float) doubleValue + "')", FLOAT, (float) Math.atan((float) doubleValue));
         }
-        assertFunction("atan(NULL)", DOUBLE, null);
+        assertFunction("atan(NULL)", FLOAT, null);
+        assertFunction("atan(CAST(NULL AS DOUBLE))", DOUBLE, null);
     }
 
     @Test
     public void testAtan2()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("atan2(" + doubleValue + ", " + doubleValue + ")", DOUBLE, Math.atan2(doubleValue, doubleValue));
-            assertFunction("atan2(FLOAT '" + (float) doubleValue + "', FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.atan2((float) doubleValue, (float) doubleValue));
+            assertFunction("atan2(FLOAT '" + (float) doubleValue + "', FLOAT '" + (float) doubleValue + "')", FLOAT, (float) Math.atan2((float) doubleValue, (float) doubleValue));
```

Comment:
Maybe we should also add test of mixed `atan(float, double)`? It should coerce float to double and call `atan(double, double)`.